Plugin/Model/OrderRepositoryPlugin.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Ampersand\DisableStockReservation\Plugin\Model;
+
+use Magento\Sales\Model\OrderRepository;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\Data\OrderSearchResultInterface;
+use Magento\Sales\Api\Data\OrderExtensionFactory;
+use Ampersand\DisableStockReservation\Model\SourcesRepository;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterfaceFactory;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterface;
+
+/**
+ * Class OrderRepositoryPlugin
+ * @package Ampersand\DisableStockReservation\Plugin\Model
+ */
+class OrderRepositoryPlugin
+{
+    /**
+     * @var SourcesRepository
+     */
+    private $sourcesRepository;
+
+    /**
+     * @var OrderExtensionFactory
+     */
+    private $orderExtensionFactory;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultInterfaceFactory;
+
+    /**
+     * OrderRepositoryPlugin constructor.
+     * @param SourcesRepository $sourcesRepository
+     * @param OrderExtensionFactory $orderExtensionFactory
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultInterfaceFactory
+     */
+    public function __construct(
+        SourcesRepository $sourcesRepository,
+        OrderExtensionFactory $orderExtensionFactory,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultInterfaceFactory
+    ) {
+        $this->sourcesRepository = $sourcesRepository;
+        $this->orderExtensionFactory = $orderExtensionFactory;
+        $this->sourceSelectionResultInterfaceFactory = $sourceSelectionResultInterfaceFactory;
+    }
+
+    /**
+     * @param OrderRepository $subject
+     * @param OrderInterface $result
+     * @return OrderInterface
+     */
+    public function afterGet(OrderRepository $subject, OrderInterface $result): OrderInterface
+    {
+        return $this->applyExtensionAttributesToOrder($result, null);
+    }
+
+    /**
+     * @param OrderRepository $subject
+     * @param OrderSearchResultInterface $result
+     * @return OrderSearchResultInterface
+     */
+    public function afterGetList(OrderRepository $subject, OrderSearchResultInterface $result): OrderSearchResultInterface
+    {
+        $sourcesCollection = $this->sourcesRepository->getOrdersSourcesCollection();
+        foreach ($result->getItems() as $item) {
+            $sourceSelectionResult = $this->sourceSelectionResultInterfaceFactory->create([
+                'sourceItemSelections' => $sourcesCollection->addFieldToFilter('order_id', $item->getEntityId())->getData(),
```

Comment:
So I think calling `getData` here would do the exact same as `load`? I could be wrong, however we're still querying the DB inside a loop. You need to do something along the lines of:
- Query your extension table will all the ID's from `getItems` (this is where a collection comes in handy, build your collection first with all the required extension table entries)
- Index your collection by order ID (so iterate it, create an array or similar with order ID keys => collection item)
- Iterate `$result->getItems()` and check if an extension table entry exists in your indexed order ID's
- Apply whatever logic you need to the currently iterated item based on the indexed data